components/camel-kubernetes/src/main/java/org/apache/camel/component/kubernetes/secrets/vault/SecretsReloadTriggerTask.java

```diff
@@ -0,0 +1,162 @@
+    public void setReloadEnabled(boolean reloadEnabled) {
+        this.reloadEnabled = reloadEnabled;
+    }
+
+    @Override
+    protected void doStart() throws Exception {
+        super.doStart();
+
+        // auto-detect secrets in-use
+        PropertiesComponent pc = camelContext.getPropertiesComponent();
+        PropertiesFunction pf = pc.getPropertiesFunction("secret");
+        if (pf instanceof SecretPropertiesFunction) {
+            propertiesFunction = (SecretPropertiesFunction) pf;
+            LOG.debug("Auto-detecting secrets from properties-function: {}", pf.getName());
+        }
+        // specific secrets
+        secrets = camelContext.getVaultConfiguration().kubernetes().getSecrets();
+        if (ObjectHelper.isEmpty(secrets) && propertiesFunction == null) {
+            throw new IllegalArgumentException("Secrets must be configured on Kubernetes vault configuration");
+        }
+
+        kubernetesClient = propertiesFunction.getClient();
+    }
+
+    @Override
+    protected void doShutdown() throws Exception {
+        super.doShutdown();
+
+        if (kubernetesClient != null) {
+            try {
+                kubernetesClient.close();
+            } catch (Exception e) {
+                // ignore
+            }
+            kubernetesClient = null;
+        }
+    }
+
+    @Override
+    public void run() {
+        final CountDownLatch isWatchClosed = new CountDownLatch(1);
+        Watch watch = kubernetesClient.secrets().inNamespace(kubernetesClient.getNamespace()).watch(new Watcher<>() {
+            @Override
+            public void eventReceived(Action action, Secret secret) {
+                switch (action.name()) {
+                    case "MODIFIED":
+                        if (isReloadEnabled()) {
+                            if (matchSecret(secret.getMetadata().getName())) {
+                                LOG.info("Update for Kubernetes Secret: {} detected, triggering CamelContext reload",
+                                        secret.getMetadata().getName());
+                                ContextReloadStrategy reload = camelContext.hasService(ContextReloadStrategy.class);
+                                if (reload != null) {
+                                    // trigger reload
+                                    reload.onReload(this);
+                                }
+                            }
+                        }
+                        break;
+                    default:
+                        LOG.debug("Not watched event {}", action.name());
+                }
+            }
+
+            @Override
+            public void onClose(WatcherException e) {
+                isWatchClosed.countDown();
+            }
+        });
+
+        // Wait till watch gets closed
+        try {
+            isWatchClosed.await();
+        } catch (InterruptedException e) {
+            LOG.error("Interrupted while waiting for the watch to close: {}", e.getMessage());
```

Comment:
Not sure if this exception brings value to end users, so it can be DEBUG level. Or at most WARN